packages/dds/tree/src/test/shared-tree/editing.spec.ts

```diff
@@ -3,135 +3,159 @@
  * Licensed under the MIT License.
  */
 import { strict as assert } from "assert";
-import { singleTextCursor } from "../../feature-libraries";
-import { jsonString } from "../../domains";
-import { moveToDetachedField, rootFieldKeySymbol, UpPath } from "../../core";
-import { brand, JsonCompatible } from "../../util";
+import { MockFluidDataStoreRuntime } from "@fluidframework/test-runtime-utils";
+import { FieldKinds, singleTextCursor } from "../../feature-libraries";
+import { jsonSchemaData, jsonString, singleJsonCursor } from "../../domains";
+import { rootFieldKeySymbol, UpPath, fieldSchema, rootFieldKey, SchemaData } from "../../core";
+import { JsonCompatible, brand } from "../../util";
 import { fakeRepair } from "../utils";
+import { ISharedTree, ISharedTreeBranch, SharedTreeFactory } from "../../shared-tree";
 import { Sequencer, TestTree, TestTreeEdit } from "./testTree";
 
+const factory = new SharedTreeFactory();
+const runtime = new MockFluidDataStoreRuntime();
+// For now, require tree to be a list of strings.
+const schema: SchemaData = {
+	treeSchema: jsonSchemaData.treeSchema,
+	globalFieldSchema: new Map([
+		[rootFieldKey, fieldSchema(FieldKinds.sequence, [jsonString.name])],
+	]),
+};
+
+function makeTree(...json: string[]): ISharedTree {
+	const tree = factory.create(runtime, "TestSharedTree");
+	tree.storedSchema.update(schema);
+	const field = tree.editor.sequenceField(undefined, rootFieldKeySymbol);
+	field.insert(0, json.map(singleJsonCursor));
+	return tree;
+}
+
 describe("Editing", () => {
 	describe("Sequence Field", () => {
 		it("can order concurrent inserts within concurrently deleted content", () => {
-			const sequencer = new Sequencer();
-			const tree1 = TestTree.fromJson(["A", "B", "C", "D"]);
-			const tree2 = tree1.fork();
-			const tree3 = tree1.fork();
-			const tree4 = tree1.fork();
+			const tree = makeTree("A", "B", "C", "D");
+			const delAB = tree.fork();
+			const delCD = tree.fork();
+			const addX = tree.fork();
+			const addY = tree.fork();
 
 			// Make deletions in two steps to ensure that gap tracking handles comparing insertion places that
 			// were affected by different deletes.
-			const delAB = remove(tree1, 0, 2);
-			const delCD = remove(tree2, 2, 2);
-			const addX = insert(tree3, 1, "x");
-			const addY = insert(tree4, 3, "y");
+			remove(delAB, 0, 2);
+			remove(delCD, 2, 2);
+			insert(addX, 1, "x");
+			insert(addY, 3, "y");
 
-			const sequenced = sequencer.sequence([delAB, delCD, addX, addY]);
-			tree1.receive(sequenced);
-			tree2.receive(sequenced);
-			tree3.receive(sequenced);
-			tree4.receive(sequenced);
+			delAB.merge();
+			delCD.merge();
+			addX.merge();
+			addY.merge();
```

Comment:
Or would it be sufficient to compare the forest of the trunk and a branch after merging that branch?

Reply:
when a branch is merged, that consumes the branch. Our branch API (intentionally) makes it impossible to have two different versions of the same edit at the same time. If we did still have a branch to compare it would have the same  entry in the revision manager and would be inherently equal to the trunk.

Reply:
What do you think about leaving this work as a PR, marking it draft and handing it to Noah (or you, or who ever works on this next) for when further work gets done on the branch APIs?